src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeBlockFileTracer.cs

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using Nethermind.Core.Crypto;
+using Nethermind.Core;
+using Nethermind.Core.Extensions;
+using System.IO.Abstractions;
+
+namespace Nethermind.Evm.Tracing.GethStyle;
+
+public class GethLikeBlockFileTracer : BlockTracerBase<GethLikeTxTrace, GethLikeTxFileTracer>
+{
+    private const string _alphabet = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+    private readonly Block _block;
+    private Stream _file;
+    private readonly string _fileNameFormat;
+    private readonly List<string> _fileNames = new();
+    private IFileSystem _fileSystem;
+    private Utf8JsonWriter _jsonWriter;
+    private readonly GethTraceOptions _options;
+    private readonly JsonSerializerOptions _serializerOptions = new();
+
+    public GethLikeBlockFileTracer(Block block, GethTraceOptions options, IFileSystem fileSystem) : base(options?.TxHash)
```

Comment:
Don't do FileSystem here, it is so limiting! Do a `Stream` one, can be used in so many places (check first comment!)